src/esp/sensor/CameraSensor.cpp

```diff
@@ -14,62 +14,81 @@
+      "CameraSensor::CameraSensor(): The sensor sub-type is unknown", );
   // Initialize renderCamera_ first to avoid segfaults
   renderCamera_ = new gfx::RenderCamera(cameraNode);
-  setProjectionParameters(spec);
+  setProjectionParameters(*spec);
   renderCamera_->setAspectRatioPolicy(
       Mn::SceneGraph::AspectRatioPolicy::Extend);
   recomputeProjectionMatrix();
   renderCamera_->setViewport(this->framebufferSize());
 }  // ctor
 
-void CameraSensor::setProjectionParameters(const SensorSpec::ptr& spec) {
-  ASSERT(spec != nullptr);
+void CameraSensor::setProjectionParameters(const CameraSensorSpec& spec) {
   // update this sensor's sensor spec to reflect the passed new values
-  spec_->resolution = spec->resolution;
-  for (const auto& elem : spec->parameters) {
-    spec_->parameters.at(elem.first) = elem.second;
-  }
-
-  width_ = spec_->resolution[1];
-  height_ = spec_->resolution[0];
-  near_ = std::atof(spec_->parameters.at("near").c_str());
-  far_ = std::atof(spec_->parameters.at("far").c_str());
-  setCameraType(spec->sensorSubType);
+  cameraSensorSpec_->resolution = spec.resolution;
+  setCameraType(spec.sensorSubType);
 
 }  // setProjectionParameters
 
 void CameraSensor::recomputeProjectionMatrix() {
   projectionMatrix_ = zoomMatrix_ * baseProjMatrix_;
   // update renderCamera_'s projectionMatrix every time the sensor's
   // projectionMatrix changes
-  renderCamera_->setProjectionMatrix(width_, height_, projectionMatrix_);
+  renderCamera_->setProjectionMatrix(cameraSensorSpec_->resolution[1],
+                                     cameraSensorSpec_->resolution[0],
+                                     projectionMatrix_);
 }
 
 void CameraSensor::recomputeBaseProjectionMatrix() {
   // refresh size after relevant parameters have changed
   Mn::Vector2 nearPlaneSize_ =
-      Mn::Vector2{1.0f, static_cast<float>(height_) / width_};
+      Mn::Vector2{1.0f, static_cast<float>(cameraSensorSpec_->resolution[0]) /
+                            cameraSensorSpec_->resolution[1]};
   float scale;
-  if (spec_->sensorSubType == SensorSubType::Orthographic) {
-    scale = std::atof(spec_->parameters.at("ortho_scale").c_str());
+  if (cameraSensorSpec_->sensorSubType == SensorSubType::Orthographic) {
+    scale = cameraSensorSpec_->ortho_scale;
     nearPlaneSize_ /= scale;
     baseProjMatrix_ =
         Mn::Matrix4::orthographicProjection(nearPlaneSize_, near_, far_);
   } else {
-    if (spec_->sensorSubType != SensorSubType::Pinhole) {
+    if (cameraSensorSpec_->sensorSubType != SensorSubType::Pinhole) {
       LOG(INFO) << "CameraSensor::setCameraType : Unsupported Camera type val :"
-                << static_cast<int>(spec_->sensorSubType)
+                << static_cast<int>(cameraSensorSpec_->sensorSubType)
                 << " so defaulting to Pinhole.";
-      spec_->sensorSubType = SensorSubType::Pinhole;
+      cameraSensorSpec_->sensorSubType = SensorSubType::Pinhole;
```

Comment:
This is the fallback which should be removed after adding the CORRADE_ASSERT mentioned earlier.